python/pyspark/sql/types.py

```diff
@@ -71,6 +67,15 @@ def json(self):
                           separators=(',', ':'),
                           sort_keys=True)
 
+    def needSerialization(self):
```

Comment:
This is expected to be override(also may be expensive), I think it's better to be a method, instead of property.